### ISSUE TITLE
Kiali with GKE-installed Istio addon not working : Workload istio-security-post-install-1.4.10-gke.5 is not found as Job

### ISSUE THREAD

**Issue by didier-durand** [kiali/kiali#3218]
Hello,

**General remarks**

I am trying to run Kiali with Istio installed automatically as add-on of a GKE cluster:

gcloud beta container clusters create "$GKE_CLUSTER" \
      --cluster-version="$GKE_VERSION" \
      --addons=Istio \
      --istio-config=auth='MTLS_PERMISSIVE'
     etc....

While I have 10 services up and running (1'000s of requests answered) the service graph remains empty.

It seems from the logs of the pod that Kiali is looking for some specific job (probably not run or deleted by the GKE customization)

I0916 08:48:57.479177       1 kiali.go:66] Kiali: Version: v1.22.1, Commit: eb8fce019ab5b31db4a414133dc599172e9634ce
I0916 08:48:57.480895       1 kiali.go:205] Using authentication strategy [anonymous]
W0916 08:48:57.480991       1 kiali.go:213] Kiali auth strategy is configured for anonymous access - users will not be authenticated.
I0916 08:48:57.481087       1 kiali.go:87] Kiali: Console version: 1.22.1
I0916 08:48:57.481164       1 kiali.go:284] Updating base URL in index.html with [/kiali]
I0916 08:48:57.481898       1 kiali.go:265] Generating env.js from config
I0916 08:48:57.490763       1 server.go:67] Server endpoint will start at [:20001/kiali]
I0916 08:48:57.492849       1 server.go:68] Server endpoint will serve static content from [/opt/kiali/console]
I0916 08:48:57.494507       1 metrics_server.go:18] Starting Metrics Server on [:9090]
I0916 09:31:30.372080       1 cache.go:126] Kiali Cache is active for namespaces [.*]
I0916 09:31:30.433988       1 cache.go:161] Waiting for Kiali cache for [namespace: istio-system] to sync
I0916 09:31:30.534247       1 cache.go:174] Kiali cache for [namespace: istio-system] started
I0916 09:31:30.534704       1 cache.go:161] Waiting for Kiali cache for [namespace: default] to sync
I0916 09:31:30.635008       1 cache.go:174] Kiali cache for [namespace: default] started
E0916 09:31:30.761033       1 workloads.go:625] Workload istio-security-post-install-1.4.10-gke.5 is not found as Job
E0916 09:31:30.851563       1 workloads.go:625] Workload istio-security-post-install-1.4.10-gke.5 is not found as Job
E0916 09:31:38.850577       1 workloads.go:625] Workload istio-security-post-install-1.4.10-gke.5 is not found as Job
E0916 09:31:39.503470       1 workloads.go:625] Workload istio-security-post-install-1.4.10-gke.5 is not found as Job
E0916 09:31:39.518349       1 workloads.go:625] Workload istio-security-post-install-1.4.10-gke.5 is not found as Job

**Describe the bug**

The graph of the services remains empty while there is significant traffic among the services.

**Versions used**
Kiali: 1.22.1
Istio: 1.4.10 (for GKE)
Kubernetes flavour and version: 1.17.9-gke.1504'

**To Reproduce**

create a cluster on GKE with same version & and activate Kiali via kubectl apply -f https://raw.githubusercontent.com/istio/istio/master/samples/addons/kiali.yaml

**Expected behavior**

Have a services graph where we can see effective traffic

**Comment by jotak** [comment]
Hi @didier-durand 
Could you open the prometheus console and check if istio metrics are present? For instance, look for the metric `istio_requests_total`.
While the error displayed in logs is perhaps a problem, I doubt it is the reason why you don't see anything in the graph. There's probably something else.
Also, do you see any error reported in the UI? Or in the javascript console?

**Comment by lucasponce** [comment]
@didier-durand 
I think that Kiali doesn't fetch Jobs by default.
Would you mind to adjust that in the config ?

Not sure if you install Kiali via Operator (then use CR) or Helm (then just update that in the ConfigMap and restart the Kiali Pod).
https://github.com/kiali/kiali-operator/blob/master/deploy/kiali/kiali_cr.yaml#L724

**Comment by didier-durand** [comment]
Guys,  

I just found this one which may be interesting : https://github.com/istio/istio/issues/25889
This is exactly what I am doing and the results that I reach.

@jotak : will check and report
@lucasponce : I wiil  check if the install via simplified addon is reponsible for cheking the job and update my config accordingly

**Comment by didier-durand** [comment]
@jotak & @lucasponce: I believe that I found the issue when running on GKE: if you do 'gcloud container clusters create' with --addons=Istio, the cluster is created with an istio-system namespace and a running prometheus instance (with an additional prom-sd pod forwarding data to stackdriver (sd). But, this prometheus instance is not configured at all like what Kiali expects to function properly.

So, I started my cluster without the addon and added Kiali as per Istio doc: it works now fine.

The issue is now that the absence of GCP prometheus and prom-sd drastically reduces the features in GCP console. But, it's not a concern for Kiali: I'll work it out with GCP folks. 

So, this ticket can be closed.

Thanks for the hints that have helped to find the issue.

Didier
